JDK 23 Build failed

## Description ##

The build is failing with JDK 23.

## Steps to reproduce ##
1. Use JDK 23
2. Run mvn verify
3. Build failed with errors:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project common: Compilation failure: Compilation failure: 
[ERROR] /Users/ishche/projects/che-che4z-lsp-for-cobol/server/common/src/main/java/org/eclipse/lsp/cobol/common/AnalysisConfig.java:[49,12] constructor AnalysisConfig in class org.eclipse.lsp.cobol.common.AnalysisConfig cannot be applied to given types;
[ERROR]   required: no arguments
[ERROR]   found:    org.eclipse.lsp.cobol.common.copybook.CopybookProcessingMode,com.google.common.collect.ImmutableList<java.lang.Object>,boolean,boolean,com.google.common.collect.ImmutableList<java.lang.Object>,com.google.common.collect.ImmutableMap<java.lang.String,com.google.gson.JsonElement>
[ERROR]   reason: actual and formal argument lists differ in length
[ERROR] /Users/ishche/projects/che-che4z-lsp-for-cobol/server/common/src/main/java/org/eclipse/lsp/cobol/common/AnalysisConfig.java:[58,12] constructor AnalysisConfig in class org.eclipse.lsp.cobol.common.AnalysisConfig cannot be applied to given types;
[ERROR]   required: no arguments
[ERROR]   found:    org.eclipse.lsp.cobol.common.copybook.CopybookProcessingMode,com.google.common.collect.ImmutableList<java.lang.Object>,boolean,boolean,com.google.common.collect.ImmutableList<java.lang.Object>,com.google.common.collect.ImmutableMap<java.lang.String,com.google.gson.JsonElement>
[ERROR]   reason: actual and formal argument lists differ in length
[ERROR] /Users/ishche/projects/che-che4z-lsp-for-cobol/server/common/src/main/java/org/eclipse/lsp/cobol/common/copybook/CopybookProcessingMode.java:[46,21] non-static method isFileUnderExtendedSourceFolder(java.lang.String) cannot be referenced from a static context
[ERROR] /Users/ishche/projects/che-che4z-lsp-for-cobol/server/common/src/main/java/org/eclipse/lsp/cobol/common/AnalysisResult.java:[36,60] cannot find symbol
...
```

## Expected behavior ##

Build is successful

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -11,7 +11,8 @@
 #   Broadcom, Inc. - initial API and implementation
 
 name: build
-
+env:
+  graalvm-version: '21'
 on:
   workflow_dispatch:
     inputs:
@@ -84,7 +85,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: graalvm/setup-graalvm@v1
         with:
-          java-version: '17'
+          java-version: ${{ env.graalvm-version }}
           components: 'native-image'
           distribution: 'graalvm-community'
           github-token: ${{ secrets.GITHUB_TOKEN }}
@@ -242,7 +243,7 @@ jobs:
       - uses: graalvm/setup-graalvm@v1
         if: matrix.os == 'ubuntu-latest' && matrix.arch == 'x64'
         with:
-          java-version: '17'
+          java-version: ${{ env.graalvm-version }}
           components: 'native-image'
           distribution: 'graalvm-community'
           native-image-musl: 'true'
@@ -253,7 +254,7 @@ jobs:
       - uses: graalvm/setup-graalvm@v1
         if: matrix.os != 'ubuntu-latest' || matrix.arch != 'x64'
         with:
-          java-version: '17'
+          java-version: ${{ env.graalvm-version }}
           components: 'native-image'
           distribution: 'graalvm-community'
           github-token: ${{ secrets.GITHUB_TOKEN }}

--- a/server/common/pom.xml
+++ b/server/common/pom.xml
@@ -26,16 +26,18 @@
     <properties>
         <maven.compiler.source>8</maven.compiler.source>
         <maven.compiler.target>8</maven.compiler.target>
-        <lombok.version>1.18.30</lombok.version>
+        <lombok.version>1.18.36</lombok.version>
+        <maven.compiler.plugin.version>3.13.0</maven.compiler.plugin.version>
         <google.guava.version>33.2.1-jre</google.guava.version>
+        <gson.version>2.11.0</gson.version>
         <logback.classic.version>1.3.14</logback.classic.version>
         <lsp4j.version>0.14.0</lsp4j.version>
         <commons.lang.version>3.12.0</commons.lang.version>
         <junit-jupiter.version>5.6.0</junit-jupiter.version>
         <junit.platform.version>1.6.0</junit.platform.version>
-        <mockito.core.version>5.10.0</mockito.core.version>
+        <mockito.core.version>5.14.2</mockito.core.version>
         <maven.surefire.plugin.version>3.0.0-M3</maven.surefire.plugin.version>
-        <maven.jacoco.plugin.version>0.8.11</maven.jacoco.plugin.version>
+        <maven.jacoco.plugin.version>0.8.12</maven.jacoco.plugin.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     </properties>
@@ -77,6 +79,10 @@
     <profiles>
         <profile>
             <id>release</id>
+            <properties>
+                <maven.compiler.source>8</maven.compiler.source>
+                <maven.compiler.target>8</maven.compiler.target>
+            </properties>
             <build>
                 <plugins>
                     <plugin>
@@ -144,7 +150,7 @@
         <dependency>
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
-            <version>2.10</version>
+            <version>${gson.version}</version>
         </dependency>
         <dependency>
             <groupId>ch.qos.logback</groupId>
@@ -202,6 +208,14 @@
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven.compiler.plugin.version}</version>
+                <configuration>
+                    <proc>full</proc>
+                </configuration>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>

--- a/server/dialect-daco/pom.xml
+++ b/server/dialect-daco/pom.xml
@@ -32,15 +32,16 @@
         <maven.compiler.target>8</maven.compiler.target>
         <maven.antlr4.plugin.version>4.13.2</maven.antlr4.plugin.version>
         <antlr.version>4.13.2</antlr.version>
-        <lombok.version>1.18.30</lombok.version>
+        <lombok.version>1.18.36</lombok.version>
+        <maven.compiler.plugin.version>3.13.0</maven.compiler.plugin.version>
         <maven.resources.plugin.version>3.0.2</maven.resources.plugin.version>
         <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
         <junit-jupiter.version>5.6.0</junit-jupiter.version>
         <junit.platform.version>1.6.0</junit.platform.version>
-        <mockito.core.version>5.10.0</mockito.core.version>
+        <mockito.core.version>5.14.2</mockito.core.version>
         <maven.cobertura.plugin.version>2.7</maven.cobertura.plugin.version>
         <maven.coveralls.plugin.version>4.3.0</maven.coveralls.plugin.version>
-        <maven.jacoco.plugin.version>0.8.11</maven.jacoco.plugin.version>
+        <maven.jacoco.plugin.version>0.8.12</maven.jacoco.plugin.version>
         <maven.surefire.plugin.version>3.0.0-M3</maven.surefire.plugin.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
@@ -143,6 +144,14 @@
         </resources>
 
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven.compiler.plugin.version}</version>
+                <configuration>
+                    <proc>full</proc>
+                </configuration>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>

--- a/server/dialect-idms/pom.xml
+++ b/server/dialect-idms/pom.xml
@@ -33,13 +33,14 @@
         <maven.compiler.target>8</maven.compiler.target>
         <maven.antlr4.plugin.version>4.13.2</maven.antlr4.plugin.version>
         <antlr.version>4.13.2</antlr.version>
-        <lombok.version>1.18.30</lombok.version>
+        <lombok.version>1.18.36</lombok.version>
         <maven.resources.plugin.version>3.0.2</maven.resources.plugin.version>
         <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
         <junit-jupiter.version>5.6.0</junit-jupiter.version>
         <junit.platform.version>1.6.0</junit.platform.version>
-        <mockito.core.version>5.10.0</mockito.core.version>
-        <maven.jacoco.plugin.version>0.8.11</maven.jacoco.plugin.version>
+        <mockito.core.version>5.14.2</mockito.core.version>
+        <maven.compiler.plugin.version>3.13.0</maven.compiler.plugin.version>
+        <maven.jacoco.plugin.version>0.8.12</maven.jacoco.plugin.version>
         <maven.surefire.plugin.version>3.0.0-M3</maven.surefire.plugin.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
@@ -168,6 +169,14 @@
         </resources>
 
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven.compiler.plugin.version}</version>
+                <configuration>
+                    <proc>full</proc>
+                </configuration>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>

--- a/server/engine/pom.xml
+++ b/server/engine/pom.xml
@@ -30,6 +30,8 @@
     </parent>
 
     <properties>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
         <antlr.version>4.13.2</antlr.version>
         <awaitility.version>3.0.0</awaitility.version>
         <commons.io.version>2.14.0</commons.io.version>
@@ -39,20 +41,19 @@
         <exclude.tests>nothing.to.exclude</exclude.tests>
         <google.guava.version>33.2.1-jre</google.guava.version>
         <guice.version>4.2.2</guice.version>
-        <java.version>8</java.version>
         <junit-jupiter.version>5.6.0</junit-jupiter.version>
         <junit.platform.version>1.6.0</junit.platform.version>
         <logback.classic.version>1.3.14</logback.classic.version>
-        <lombok.version>1.18.30</lombok.version>
+        <lombok.version>1.18.36</lombok.version>
         <lsp4j.version>0.14.0</lsp4j.version>
         <picocli.version>4.7.5</picocli.version>
         <maven.antlr4.plugin.version>4.13.2</maven.antlr4.plugin.version>
         <maven.assembly.plugin.version>3.2.0</maven.assembly.plugin.version>
-        <maven.compiler.plugin.version>3.7.0</maven.compiler.plugin.version>
-        <maven.jacoco.plugin.version>0.8.11</maven.jacoco.plugin.version>
+        <maven.compiler.plugin.version>3.13.0</maven.compiler.plugin.version>
+        <maven.jacoco.plugin.version>0.8.12</maven.jacoco.plugin.version>
         <maven.resources.plugin.version>3.0.2</maven.resources.plugin.version>
         <maven.surefire.plugin.version>3.0.0-M3</maven.surefire.plugin.version>
-        <mockito.core.version>5.10.0</mockito.core.version>
+        <mockito.core.version>5.14.2</mockito.core.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <slf4j.api.version>2.0.9</slf4j.api.version>
@@ -244,8 +245,7 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>${maven.compiler.plugin.version}</version>
                 <configuration>
-                    <source>${java.version}</source>
-                    <target>${java.version}</target>
+                    <proc>full</proc>
                 </configuration>
             </plugin>
             <plugin>

--- a/server/parser/pom.xml
+++ b/server/parser/pom.xml
@@ -29,7 +29,8 @@
     </parent>
 
     <properties>
-        <java.version>8</java.version>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>8</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 
@@ -39,9 +40,9 @@
         <lifecycle-mapping.plugin.version>1.0.0</lifecycle-mapping.plugin.version>
         <maven.resources.plugin.version>3.0.2</maven.resources.plugin.version>
         <maven.antlr4.plugin.version>4.13.2</maven.antlr4.plugin.version>
-        <maven.compiler.plugin.version>3.7.0</maven.compiler.plugin.version>
+        <maven.compiler.plugin.version>3.13.0</maven.compiler.plugin.version>
         <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
-        <maven.jacoco.plugin.version>0.8.11</maven.jacoco.plugin.version>
+        <maven.jacoco.plugin.version>0.8.12</maven.jacoco.plugin.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     </properties>
@@ -133,8 +134,7 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>${maven.compiler.plugin.version}</version>
                 <configuration>
-                    <source>${java.version}</source>
-                    <target>${java.version}</target>
+                    <proc>full</proc>
                 </configuration>
             </plugin>
             <plugin>

--- a/server/test/pom.xml
+++ b/server/test/pom.xml
@@ -17,12 +17,13 @@
         <maven.antlr4.plugin.version>4.13.2</maven.antlr4.plugin.version>
         <antlr.version>4.13.2</antlr.version>
         <lombok.version>1.18.30</lombok.version>
+        <maven.compiler.plugin.version>3.13.0</maven.compiler.plugin.version>
         <maven.resources.plugin.version>3.0.2</maven.resources.plugin.version>
         <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
         <junit-jupiter.version>5.6.0</junit-jupiter.version>
         <junit.platform.version>1.6.0</junit.platform.version>
-        <mockito.core.version>5.10.0</mockito.core.version>
-        <maven.jacoco.plugin.version>0.8.11</maven.jacoco.plugin.version>
+        <mockito.core.version>5.14.2</mockito.core.version>
+        <maven.jacoco.plugin.version>0.8.12</maven.jacoco.plugin.version>
         <maven.surefire.plugin.version>3.0.0-M3</maven.surefire.plugin.version>
         <guice.version>4.2.2</guice.version>
         <autoservice.version>1.0.1</autoservice.version>
@@ -157,6 +158,14 @@
         </resources>
 
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven.compiler.plugin.version}</version>
+                <configuration>
+                    <proc>full</proc>
+                </configuration>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>

